Library/Homebrew/dev-cmd/update-python-resources.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require "cli/parser"
+
+module Homebrew
+  module_function
+
+  def update_python_resources_args
+    Homebrew::CLI::Parser.new do
+      usage_banner <<~EOS
+        `update-python-resources` [<options>] <formula>
+
+        Update versions for PyPI resource blocks in <formula>.
+      EOS
+      switch "-p", "--print-only",
+             description: "Print the updated resource blocks instead of changing the formula."
+      named :formula
+    end
+  end
+
+  def update_python_resources
+    update_python_resources_args.parse
+
+    formula = args.formulae.first
+
+    pypi_resources = formula.resources.select do |resource|
+      resource.url.match?(%r{^https?://files.pythonhosted.org/packages/})
+    end
+
+    odie "No PyPI resources found in \"#{formula.name}\"" if pypi_resources.empty?
+
+    odie "\"python@3.8\" must be installed" unless Formula["python@3.8"].any_version_installed?
+
+    unless File.exist?("#{HOMEBREW_PREFIX}/bin/virtualenv")
+      odie "Please install \"virtualenv\" by running `pip3 install virtualenv`"
+    end
+
+    Mktemp.new("poet-env").run do
+      ohai "Temporarily installing \"#{formula.name}\"" unless args.print_only?
+
+      # Create virtualenv and install package and poet
+      Utils.safe_popen_read "#{HOMEBREW_PREFIX}/bin/virtualenv", "venv"
+      Utils.safe_popen_read "venv/bin/pip", "install", formula.name, "homebrew-pypi-poet"
+
+      # Run poet
+      venv_pythonpath = "venv/lib/python3.8/site-packages"
+      poet_output = Utils.safe_popen_read({ "PYTHONPATH" => venv_pythonpath }, "venv/bin/poet", formula.name)
+
+      # Remove formula name resource from poet output
+      poet_output.sub!(/^\s*resource "snakemake" do\s*url .*\s*sha256 .*\s*end\n/, "")
```

Comment:
I think you want `poet_output.sub!(/^\s*resource "#{formula.name}" do\s*url .*\s*sha256 .*\s*end\n/, "")` here.

Reply:
🤦 Whoops... Yep, I was using `snakemake` for testing